### DISCUSSION TITLE
Simple way to reject bool when int is required?

### DISCUSSION THREAD

**Discussion by pchanial** [wyfo/apischema#379] [Q&A]
In Python, `bool` is a subclass of `int`.
Is there a simple way to exclude subclasses (other than checking in an ad-hoc deserializer) so that the following deserialization raises a validation error?
```python
@dataclass
class Foo:
    value: int

deserialize(Foo, {'value': True})
```

**Comment by wyfo** [comment]
I think this is simply a design flaw in *apischema*: integer should not allow boolean values. I will add an extra check in the code.

EDIT: Fixed in #385
